src/operator/tensor/pseudo2DTranspose_op-inl.cuh

```diff
@@ -78,23 +85,34 @@ __global__ void transpose_pseudo2D(DType* out, DType* inp,
         }
         __syncthreads();
 
-        // read from shared to registers
-        transp_t tmp[TSR];
+        // read from shared to local registers
+        CType tmp[TSR];
         #pragma unroll
         for (index_t i = 0; i < TSR; i++) {
+          DType* tmp_dptr = reinterpret_cast<DType*>(&tmp[i]);
```

Comment:
Hmmm, I'm not sure if this will still be handled properly by the compiler, let me test that.

Reply:
Have you find any problem? I can compile that and it also passed CI.

Reply:
Hi, I only now had a chance to look into it. No, there is not any problem - I was worried that the compiler could get confused by this and put the `tmp` array in local memory instead of registers, but I tested and it does not do it.

Reply:
Great! Thanks for clarifying 👍